prow/apis/prowjobs/v1/types.go

```diff
@@ -220,6 +313,42 @@ type GCSConfiguration struct {
 	DefaultRepo string `json:"default_repo,omitempty"`
 }
 
+func (g *GCSConfiguration) ApplyDefault(def *GCSConfiguration) *GCSConfiguration {
+	if g == nil {
+		return def
```

Comment:
Given that these are used all over the place, I believe it will be much safer to clone def and return the pointer to the cloned object, rather than allowing plugin foo to overwrite the configuration here.

Reply:
Done.